File: MiniDOMBase/DMD/Anagrafica/ListeRicontatto/Importers/XLSImporterTable.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports System
Imports minidom.Net.Mail
Imports minidom.Anagrafica
Imports minidom.XML

Partial Public Class Anagrafica

    <Serializable>
    Public Class XLSImporterTable
        Implements XML.IDMDXMLSerializable

        Public Name As String
        Public Columns As CCollection(Of XLSImporterColumn)

        Public Sub New()
            Me.Name = ""
            Me.Columns = New CCollection(Of XLSImporterColumn)
        End Sub

        Protected Overridable Sub XMLSerialize(writer As XMLWriter) Implements IDMDXMLSerializable.XMLSerialize
            writer.WriteAttribute("Name", Me.Name)
            writer.WriteTag("Columns", Me.Columns)
        End Sub

        Protected Overridable Sub SetFieldInternal(fieldName As String, fieldValue As Object) Implements IDMDXMLSerializable.SetFieldInternal
            Select Case fieldName
                Case "Name" : Me.Name = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "Columns" : Me.Columns.AddRange(fieldValue)
            End Select
        End Sub
    End Class


End Class

File: MiniDOMBase/DMD/Anagrafica/ListeRicontatto/Importers/XLSImporterTable.cs
using System;
using System.Collections;
using DMD.XML;
using DMD.Databases.Collections;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Tabella importata
        /// </summary>
        [Serializable]
        public class XLSImporterTable
            : IDMDXMLSerializable
        {

            /// <summary>
            /// Nome
            /// </summary>
            public string Name;

            /// <summary>
            /// Colonne
            /// </summary>
            public CCollection<XLSImporterColumn> Columns;


            /// <summary>
            /// Costruttore
            /// </summary>
            public XLSImporterTable()
            {
                Name = "";
                Columns = new CCollection<XLSImporterColumn>();
            }

            /// <summary>
            /// Serializzazione XML
            /// </summary>
            /// <param name="writer"></param>
            protected virtual void XMLSerialize(XMLWriter writer)
            {
                writer.WriteAttribute("Name", Name);
                writer.WriteTag("Columns", Columns);
            }

            /// <summary>
            /// Deserializzazione XML
            /// </summary>
            /// <param name="fieldName"></param>
            /// <param name="fieldValue"></param>
            protected virtual void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "Name":
                        {
                            Name = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case "Columns":
                        {
                            Columns.AddRange((IEnumerable)fieldValue);
                            break;
                        }
                }
            }

            void IDMDXMLSerializable.XMLSerialize(XMLWriter writer)
            {
                this.XMLSerialize(writer);
            }

            void IDMDXMLSerializable.SetFieldInternal(string fieldName, object fieldValue)
            {
                this.SetFieldInternal(fieldName, fieldValue);
            }
        }
    }
}